stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,436 @@
+//===--- android.modulemap ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// The module map for the Android NDK.
+// The majority of the modules in this file are prefixed
+// with an underscore, to discourage users from importing them from
+// Swift directly. Instead, users should import 'Android' or another
+// umbrella module that includes these headers.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The top level 'Bionic' module is the C standard library
+// used in the Android NDK.
+module Bionic [system] {
+  header "SwiftBionic.h"
+  export *
+}
+
+// The individual modules for the Bionic C standard library used
+// by the Android NDK.
+module _assert [system] {
+  // <assert.h>'s use of NDEBUG requires textual inclusion.
+  textual header "assert.h"
+}
+module _complex [system] {
+  header "complex.h"
+  export *
+}
+module _ctype [system] {
+  header "ctype.h"
+  export *
+}
+module _errno [system] {
+  header "errno.h"
+  export *
+}
+module _fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers.
+// FIXME: this can be dropped once NDK's libc++ 'std' module is split
+// into individual submodules.
+module _inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use _stdint
+  use _sys_cdefs
+}
+module _limits [system] {
+  header "limits.h"
+  header "bits/posix_limits.h"
```

Comment:
We shouldn't put multiple headers in a single module, but rather split them by submodule. Otherwise you get the confusing behavior that including one header includes the other. e.g. if some other header does `#include <bits/posix_limits.h>` then it also makes <limits.h> visible which is often not the intent. So I think we should do it like this.
```
module _limits [system] {
  header "limits.h"
  export *
  explicit module posix_limits {
    header "bits/posix_limits.h"
    export *
  }
}
```
(or if limits.h includes bits/posix_limits.h then the submodule doesn't need to be `explicit` since importing limits will import limits.posix_limits anyway - there's no harm in leaving it `explicit` though.)